### DISCUSSION TITLE
Edit function does not work properly

### DISCUSSION THREAD

**Discussion by franziska-caroline** [neuefische/questions#799] [Web] [answered]
Hello and happy new year! ✨
New Year, new problem. I am currently working on our edit function to edit the plant preferences.
At the moment, you can click on the Edit button next to the Preference Card to open a new page. There is then the FilterForm with the filterSettings you selected. You can also save it and if you log out preferenceData, where everything is saved, the new values are also there, but they are not used for reasons and the plants that match the revised preference are also not included and are undefined. 

Here is the code, we are working on the feature-edit-preferences branch:
[(https://github.com/orgs/neuefische/discussions/738#example-link-to-your-repo)](https://github.com/MartaJaekel/Plantbuddy/tree/feature-edit-preferences)

I think the error will be somewhere in our FilterForm in the handleSubmit function but I can't find it.

The goal is to get from the edit button on the preference page to the edit page, where you can edit the preference and after clicking save (or cancel) you get back to the preference page where all our preferences are listed.

Thank you for your help! 😇

**Comment by mbosselmann** [comment, marked as answer]
We discussed the problem together and the following parts needed an update:
1. `handleSubmit` function of `FilterForm` component: onEditPreference did not receive the `preferenceId` and could not update the data correctly. Solution with spread syntax:
```js
      onEditPreference({ ...preferenceData, id: preferenceId });
```

2. The form was resetted after each submit. That was the default behavior of the handleSubmit function. We did move the `reset` into the if/else statement:

```js
    if (preferenceId) {
      // Editing an existing preference
      onEditPreference({ ...preferenceData, id: preferenceId });
    } else {
      // Adding a new preference
      onAddPreference(preferenceData);
      event.target.reset();
    }
```

3. The key `plantPreferences` always got the value `undefined`.  The problem was the missing plant data that needed to be added ad prop via the `preferences/edit/[id]` page. :)

```jsx
export default function EditPreferencePage({
  preferences,
  handleEditPreference,
  plants,
}) {

...

  return (
    <>
      <StyledHeadline>PlantBuddy</StyledHeadline>;
      <main>
        <StyledTitle>Edit your Preference</StyledTitle>
        <FilterForm
          plants={plants}
          ...
        />
      </main>
    </>
  );
}

```
 Problem solved! ✨
